### PR TITLE
Default include_type_name to false in the yml test harness.

### PR DESCRIPTION
**Pull request by jtibshirani**
This PR removes the temporary change we made to the yml test harness in #37285 to automatically set `include_type_name` to `true` in index creation requests if it's not already specified. This is possible now that the vast majority of index creation requests were updated to be typeless in #37611.

Additionally, this PR updates the test harness to set `include_type_name` to `false` in index creation requests when communicating with 6.x nodes. This mirrors the logic added in #37611 to allow for typeless document write requests in test set-up code. With this update in place, we can remove many references to `include_type_name: false` from the yml tests.

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/test/search/110_field_collapsing.yml
Comment (colings86) [inline rest-api-spec/src/main/resources/rest-api-spec/test/search/110_field_collapsing.yml]: I'm curious why this is needed since in 7.0 this is the default and in mixed version 6.x/7.x tests we are injecting `include_type_name: false`?

File: test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestExecutionContext.java
Comment (colings86) [inline test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestExecutionContext.java]: Should we be checking the esVersion() here and only injecting this when we are on a mixed version cluster as in the checks below? I think it would be good to check 7.0 clusters with the default rather than injecting the parameter?

Comment (colings86) [reply test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestExecutionContext.java]: Actually nevermind. I just saw in a later commit you are doing this check

File: rest-api-spec/src/main/resources/rest-api-spec/test/suggest/30_context.yml
Comment (colings86) [inline rest-api-spec/src/main/resources/rest-api-spec/test/suggest/30_context.yml]: I think an extra new line got deleted here?

File: rest-api-spec/src/main/resources/rest-api-spec/test/search/110_field_collapsing.yml
Comment (colings86) [reply rest-api-spec/src/main/resources/rest-api-spec/test/search/110_field_collapsing.yml]: I just saw in a different commit that we only inject the parameter for the create index API so that explains why we need this here

**Review by colings86 — APPROVED** [review]
This LGTM with the commits so far up to and including https://github.com/elastic/elasticsearch/pull/38058/commits/ee9d68d617139738b2eee9955bd93f601eab0770. I suspect we will have more tests to fix though so this might need another review before we merge

**Comment by colings86** [comment]
@elasticmachine run elasticsearch-ci/2

File: rest-api-spec/src/main/resources/rest-api-spec/test/suggest/30_context.yml
Comment (jtibshirani) [reply rest-api-spec/src/main/resources/rest-api-spec/test/suggest/30_context.yml]: 👍 will add it back.

**Comment by jtibshirani** [comment]
I caught up with @colings86 offline and he was okay with merging after this latest round of fixes.

**Pull request merged**
